airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
+
+        request = conn.transferOperations().list(
+            name=TRANSFER_OPERATIONS,
+            filter=json.dumps(filter)
+        )
+
+        while request is not None:
+            response = request.execute(num_retries=NUM_RETRIES)
+            if 'operations' in response:
+                operations.extend(response['operations'])
+
+            request = conn.transferOperations().list_next(
+                previous_request=request,
+                previous_response=response)
+
+        return operations
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def pause_transfer_operation(self, operation_name):
+        """
+        Pauses an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .pause(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def resume_transfer_operation(self, operation_name):
+        """
+        Resumes an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .resume(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def wait_for_transfer_job(self,
+                              job,
+                              expected_status=[GcpTransferOperationStatus.SUCCESS]):
+        """
+        Waits until the job reaches the expected state.
+
+        :param job: Transfer job
+        :type job: dict
+        :param expected_status: State that is expected
+            See:
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferOperations#Status
+        :type expected_status: string
+        :rtype: None
+        """
         while True:
-            result = self.get_conn().transferOperations().list(
-                name='transferOperations',
-                filter=json.dumps({
-                    'project_id': job['projectId'],
-                    'job_names': [job['name']],
-                }),
-            ).execute()
-            if self._check_operations_result(result):
-                return True
+            operations = self.list_transfer_operations(filter={
+                'project_id': job['projectId'],
+                'job_names': [job['name']]
+            })
+
+            if GCPTransferServiceHook.check_operations_statuses(operations, expected_status):
+                return
             time.sleep(TIME_TO_SLEEP_IN_SECONDS)
 
-    def _check_operations_result(self, result):
-        operations = result.get('operations', [])
+    def _append_project_id(self, body, alternative):
```

Comment:
To me "append" usually means adding at the end. Here, however, we are adding the `project_id` to a dict, which doesn't guarantee any order, AFAIK.
What do you think about changing the name to `_add_...` or `_insert_...`?

Reply:
Maybe '_inject_project_id'? What do you think? It will be similar to `TransferJobPreprocessor._inject_aws_credentials`

Reply:
Yeah, `_inject_` sounds good.